### PR TITLE
Make blake2b hash util in shared/

### PR DESCRIPTION
**Pull request by IvanTheGreatDev**
Implements #646 , a shared utility function for performing blake2b hashes of data.

### CODE REVIEW

File: shared/hashutil/hash.go [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package hashutil
+
+import (
+	"golang.org/x/crypto/blake2b"
+)
+
+// Hash defines a function that returns the
```
Comment (terenc3t) [inline shared/hashutil/hash.go]: can you add this [link](https://github.com/ethereum/eth2.0-specs/blob/master/specs/beacon-chain.md#appendix-a---hash-function) in the comments? thanks

File: shared/hashutil/hash_test.go [hunk 2]
```diff
@@ -0,0 +1,17 @@
+package hashutil
```
Comment (prestonvanloon) [inline shared/hashutil/hash_test.go]: Please black box this test by using package hashutil_test and importing hashutil

File: shared/hashutil/hash_test.go [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+import (
+	"testing"
+)
+
+func TestHash(t *testing.T) {
+	hash := Hash([]byte{0})
+	test := Hash([]byte{0})
```
Comment (prestonvanloon) [inline shared/hashutil/hash_test.go]: Is there a more meaningful test you can add? Is there some known blake hash tests we could replicate?

Comment (IvanTheGreatDev) [reply shared/hashutil/hash_test.go]: I think what I changed it to is an improvement but all I can find is: https://github.com/golang/crypto/blob/master/blake2b/blake2b_test.go

Should I modify it to this or is my modification fine?

**Pull request merged**